api/citations/utils.py

```diff
@@ -75,4 +110,152 @@ def render_citation(node, style='apa'):
+
+    cit = new_apa.rstrip(', ') + ' '
+    for x in new_csl[1:]:
+        cit += '(' + x
+    return cit
+
+
+def apa_name(name):
+    apa = ''
+    if name['family_name']:
+        apa += name['family_name'] + ','
+    if name['given_name']:
+        apa += ' ' + name['given_name'][0] + '.'
+        if name['middle_names']:
+            apa += ' ' + name['middle_names'][0] + '.'
+        apa += ','
+    if name['suffix']:
+        apa += ' ' + name['suffix'] + ','
+    return apa
+
+
+def mla_reformat(node, cit):
+    contributors_list = [x for x in node.contributors if node.get_visible(x)]
+    retrive_from = cit.split('Open')[-1]
+    # throw error if there is no visible contributor
+    if len(contributors_list) == 0:
+        raise HTTPError(http.BAD_REQUEST)
+    # handle only one contributor
+    elif len(contributors_list) == 1:
+        name = process_name(node, contributors_list[0])
+        new_mla = mla_name(name, initial=True).rstrip(' ')
+    # handle more than one contributor  but less than 5 contributors
+    elif len(contributors_list) in range(1, 5):
+        first_one = mla_name(process_name(node, contributors_list[0]), initial=True)
+        rest_ones = [mla_name(process_name(node, x)) for x in contributors_list[1:-1]]
+        last_one = mla_name(process_name(node, contributors_list[-1]))
+        if rest_ones:
+            rest_part = ', '.join(rest_ones)
+            new_mla = first_one + ', ' + rest_part + ', and ' + last_one
+        else:
+            new_mla = first_one + 'and ' + last_one
+    # handle 5 or more contributors
+    else:
+        name = process_name(node, contributors_list[0])
+        new_mla = mla_name(name, initial=True) + ' et al. '
+    cit = new_mla
+    cit += u' \u201c' + node.title.title() + u'.\u201d Open' + retrive_from
+    return cit
+
+
+def chicago_reformat(node, cit):
+    new_csl = cit.split('20')
+    contributors_list = [x for x in node.contributors if node.get_visible(x)]
+
+    # throw error if there is no visible contributor
+    if len(contributors_list) == 0:
+        raise HTTPError(http.BAD_REQUEST)
+    # handle only one contributor
+    elif len(contributors_list) == 1:
+        name = process_name(node, contributors_list[0])
+        new_chi = mla_name(name, initial=True) + ' '
+    # handle more than one contributor  but less than 11 contributors
+    elif len(contributors_list) in range(1, 11):
+        first_one = mla_name(process_name(node, contributors_list[0]), initial=True)
+        rest_ones = [mla_name(process_name(node, x)) for x in contributors_list[1:-1]]
+        last_one = mla_name(process_name(node, contributors_list[-1]))
+        if rest_ones:
+            rest_part = ', '.join(rest_ones)
+            new_chi = first_one + ', ' + rest_part + ', and ' + last_one + ' '
+        else:
+            new_chi = first_one + 'and ' + last_one + ' '
+    # handle 8 or more contributors
```

Comment:
Should this be "11 or more contributors"?

Reply:
fixed in be67b43